src/app/[lang]/(protected)/chat/[chatId]/menu/change-tags-description/changeTagsDescription.tsx

```diff
@@ -0,0 +1,106 @@
+    item12: false,
+  });
+
+  const maxCharacters = 300;
+
+  const toggleDropdown = (e: any) => {
+    e.preventDefault();
+    setIsOpen(!isOpen);
+  };
+
+  const handleText = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const textValue = e.currentTarget.value.slice(0, maxCharacters);
+    setIntroduce(textValue);
+  };
+
+  const handleCheckboxChange = (e: any) => {
+    const { name, checked } = e.target;
+    setSelectedItems((prevItems) => ({
+      ...prevItems,
+      [name]: checked,
+    }));
+  };
+
+  return (
+    <div className="mt-10">
+      <div>
+        <form className="flex grow flex-col justify-between">
+          <div className="mb-10">
+            <textarea
+              value={"chat description"}
+              max-length={maxCharacters}
+              onChange={handleText}
+              className="h-[192px] w-full resize-none rounded-3xl border border-neutral-700 bg-stone-900 px-5 py-3 text-sm font-normal leading-tight text-neutral-50 outline-none focus:border-violet-700"
+              name="introduce"
+              id="introduce"
+            ></textarea>
+            <div className="px-2 text-right">
+              <span className="text-xs font-normal leading-none text-stone-300">
+                {introduce.length} / {maxCharacters}
+              </span>
+            </div>
+          </div>
+
+          <div className="relative">
+            {isOpen && (
+              <div className="absolute -top-[330px] w-full rounded-3xl border border-violet-700 bg-stone-900 p-4 transition delay-150 duration-300 ease-in-out">
+                <div className="custom-scrollbar max-h-[282px] overflow-y-scroll">
+                  {Object.entries(selectedItems).map(([key, value]) => {
+                    return (
+                      <div
+                        className={`${
+                          value ? "bg-[#252525]" : ""
+                        } w-[97%] rounded-lg px-4 py-2`}
+                        key={key}
+                      >
+                        <label className="flex items-center">
+                          <input
+                            type="checkbox"
+                            name={key}
+                            checked={value}
+                            onChange={handleCheckboxChange}
+                            className="form-checkbox hidden"
+                          />
+                          <span className="ml-2">{key}</span>
+                        </label>
+                      </div>
+                    );
+                  })}
+                </div>
+              </div>
+            )}
+
+            <div className="inline-flex h-12 w-full items-center justify-center gap-3 rounded-3xl border border-neutral-700 bg-stone-900 px-5 py-3.5">
+              <p>#selected tags</p>
+              <button onClick={toggleDropdown} className="rotate-90">
+                <Image src={ChevronRight} alt="dropdown" />
+              </button>
+            </div>
```

Comment:
Capture click events on the whole element, it would be more convenient for users

Also, capture click outside to close the menu